packages/graphql/source/fetch/stream.ts

```diff
@@ -299,15 +299,55 @@ async function* splitChunksOnBoundary(
   chunks: AsyncIterableIterator<string>,
   boundary: string,
 ) {
+  const separator = boundary + NEWLINE_SEPARATOR;
+  const encoder = new TextEncoder();
+  const decoder = new TextDecoder();
+
   let buffer = '';
   let boundaryIndex: number;
+  let headersEnd: number;
+  let contentEnd: number;
+  let rawHeaders: string;
+  let headers: Headers;
+  let contentLength;
 
   for await (const chunk of chunks) {
     buffer += chunk;
+
     while ((boundaryIndex = buffer.indexOf(boundary)) > -1) {
       yield buffer.slice(0, boundaryIndex);
       buffer = buffer.slice(boundaryIndex + boundary.length);
     }
+
+    if ((headersEnd = buffer.indexOf(HEADER_SEPARATOR)) > -1) {
+      if ((boundaryIndex = buffer.indexOf(separator)) > -1) {
+        rawHeaders = buffer.slice(boundaryIndex + separator.length, headersEnd);
+      } else {
+        rawHeaders = buffer.slice(0, headersEnd);
+      }
+
+      const emptyHeaderPairs: [string, string][] = [];
+      headers = new Headers(
+        rawHeaders.split(NEWLINE_SEPARATOR).reduce((headerPairs, element) => {
+          const [key, value, ..._] = element.split(': ');
+          if (key && value) {
+            headerPairs.push([key, value]);
+          }
+
+          return headerPairs;
+        }, emptyHeaderPairs),
+      );
+
+      if (headers.has('content-length')) {
+        contentLength = parseInt(headers.get('content-length') as string);
+        const encodedBuffer = encoder.encode(buffer);
+        const encodedSeparator = encoder.encode(HEADER_SEPARATOR);
+        headersEnd = indexOfSequence(encodedBuffer, encodedSeparator);
+        contentEnd = headersEnd + encodedSeparator.byteLength + contentLength;
+        yield decoder.decode(encodedBuffer.slice(0, contentEnd));
```

Comment:
Just realized I need to check if the byteLength of the encoded buffer is actually long enough, since the chunk might not contain all of the multipart message yet.